clang/lib/CIR/Lowering/ThroughMLIR/LowerCIRLoopToSCF.cpp

```diff
@@ -261,6 +274,40 @@ void SCFWhileLoop::transferToSCFWhileOp() {
   rewriter->eraseBlock(&scfWhileOp.getAfter().back());
 }
 
+void SCFDoLoop::transferToSCFWhileOp() {
+  // only support a simple do-while
+  // FIXME: can not support nested do-while
+
+  auto scfWhileOp = rewriter->create<mlir::scf::WhileOp>(
+      DoOp.getLoc(), DoOp->getResultTypes(), adaptor.getOperands());
```

Comment:
Could we use region builder and mergeBlocks as https://github.com/llvm/llvm-project/blob/main/mlir/lib/Dialect/SCF/Transforms/WrapInZeroTripCheck.cpp#L101?

Reply:
Nice! I will try that when free.